torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3044,10 +3044,13 @@ def replace(expr, repl):
 
         # Check if the range can solve it statically
         out = bound_sympy(new_expr, new_range_env)
-        _assert_bound_is_rational(new_expr, out)
+        try:
+            _assert_bound_is_rational(new_expr, out)
```

Comment:
Nope. Add the smallest possible exception within `assert_bound_is_rational`, please.

Reply:
Unlike the sqrt exception added already, I don't think it makes sense to add exceptions here because the expression can be anything. I added a commit that might be a better compromise.

Reply:
Compromise seems ok to me